tool/lib/utils.dart

```diff
@@ -90,22 +90,29 @@ class CliCommand {
 
   /// CliCommand helper for running git commands.
   ///
-  /// Arguments can be passed in as a single string, this helper will split them
-  /// up using their spaces. e.g. CliCommand.git(['checkout test-branch'])
+  /// Arguments can be passed in as a single string using [cmd], this will split
+  /// the string into args using spaces. e.g. CliCommand.git(cmd: 'checkout test-branch')
   ///
-  /// If you don't want the args to be split by their spaces then you can set
-  /// [split] to false.
-  factory CliCommand.git(
-    List<String> args, {
+  /// If you instead want to specify args explicitly, you can use the
+  /// [args] param. e.g. CliCommand.git(args: ['checkout', 'test-branch'])
+  factory CliCommand.git({
+    String? cmd,
+    List<String>? args,
     bool throwOnException = true,
     bool split = true,
   }) {
-    if (split) {
-      args = args.map((e) => e.split(' ')).reduce((value, element) {
-        value.addAll(element);
-        return value;
-      });
+    if ((cmd == null) && (args == null)) {
+      throw ('One of `cmd` and `args` must be specified');
+    }
+
+    if ((cmd != null) && (args != null)) {
+      throw ('Only one of `cmd` and `args`, at a time, may be specified');
```

Comment:
this can all be combined
```dart
if ((cmd == null) == (args == null)) {
  throw Exception('Exactly one of `cmd` and `args` must be specified.);
}
```